vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -133,9 +181,19 @@ def _set_categorical_selectors_options(self):
 
             self.selector.options = sorted(options)
 
+    def _set_filter_function(self):
+        if isinstance(self.selector, RangeSlider) or (
+            isinstance(self.selector, DatePicker) and self.selector.multi is True
+        ):
+            filter_function = _filter_between
+        else:
+            filter_function = _filter_isin
```

Comment:
Maybe convert this to a mapping dict as well? Similar to how we have it for the default selectors?

Reply:
I haven't tested this, so it might be broken, but something like this:

```
 filter_mapping = {
        RangeSlider: _filter_between,
        DatePicker: _filter_between if self.selector.multi else _filter_isin,
        Dropdown: _filter_isin,
        RadioItems: _filter_isin,
        Checklist: _filter_isin
    }

# Then apply this directly:
...
if not self.selector.actions:
    filter_function = filter_mapping.get(self.selector, _filter_isin)
```

Only change it if you think it's an improvement though. I am a bit indifferent right now 😄

Reply:
I would keep it simple, remove this function entirely and put the logic directly in `_set_actions`:
```
if isinstance(self.selector, RangeSlider) or isinstance(self.selector, DatePicker) and self.selector.range:
    filter_function = _filter_between
else:
    filter_function = _filter_isin
```